chaos_genius/views/download_view.py

```diff
@@ -0,0 +1,215 @@
+import csv
+import io
+
+from flask import Blueprint, Response, jsonify, request
+
+from chaos_genius.controllers.kpi_controller import get_kpi_data_from_id
+from chaos_genius.core.rca.rca_utils.api_utils import (
+    kpi_line_data,
+    rca_analysis,
+    rca_hierarchical_data,
+)
+from chaos_genius.views.anomaly_data_view import get_overall_data_points
+
+blueprint = Blueprint("downloads", __name__)
+
+
+def iter_csv(data):
+    line = io.StringIO()
+    writer = csv.writer(line, delimiter=",")
+    for csv_line in data:
+        writer.writerow(csv_line)
+        line.seek(0)
+        yield line.read()
+        line.truncate(0)
+        line.seek(0)
+
+
+@blueprint.route("/<int:kpi_id>/anomaly_data", methods=["GET"])
+def download_anomaly_data(kpi_id: int):
+    """API Endpoint to download overall KPI anomaly data in CSV form.
+
+    Data is downloaded for the last 60 days by default
+    """
+    try:
+        data_points = get_overall_data_points(kpi_id)
+        if not data_points:
+            raise Exception(f"No anomaly data found for KPI id {kpi_id}")
+
+        def row_gen(data_points):
+            csv_headers = [
+                "datetime",
+                "value",
+                "severity",
+                "upper_bound",
+                "lower_bound",
+                "series_type",
+                "series_name",
+            ]
+            yield csv_headers
+            for row in data_points:
+                attr_list = [
+                    row.data_datetime.strftime("%a %-d %B %H:%M:%S %Y"),
+                    str(row.y),
+                    str(row.severity),
+                    str(row.yhat_upper),
+                    str(row.yhat_lower),
+                    str(row.anomaly_type),
+                    str(row.series_type),
+                ]
+                yield attr_list
+
+        response = Response(iter_csv(row_gen(data_points)), mimetype="text/csv")
+        response.headers[
+            "Content-Disposition"
+        ] = f"attachment; filename=KPI-{kpi_id}-anomaly-data.csv"
+        return response
+    except Exception as e:
+        return jsonify(
+            {"status": "failure", "message": f"Downloading data failed: {e}"}
+        )
+
+
+@blueprint.route("/<int:kpi>/chart_data", methods=["GET"])
+def kpi_download_line_data(kpi: int):
+    """API endpoint to download chart data for last 60 days."""
+    try:
+        line_data = kpi_line_data(kpi, download=True)
```

Comment:
Use destructuring here to show the semantic meanings instead of just indexes:
```suggestion
        status, err_msg, data_points = kpi_line_data(kpi, download=True)
```

(the next 3 lines will also need to be changed according to this)